src/Types/Definitions/Type.php

```diff
@@ -240,9 +240,13 @@ public function isLeafType(): bool
     public function toType(): GraphQLType
     {
         $type = $this->getType();
-        $type = $this->isList() ? GraphQLType::listOf($type) : $type;
         $type = $this->isNullable() ? $type : GraphQLType::nonNull($type);
 
+        if ($this->isList()) {
+            $type = GraphQLType::listOf($type);
+            $type = $this->isNullable() ? $type : GraphQLType::nonNull($type);
```

Comment:
Wait, there are actually four possible combinations of `listOf` and `nonNull`:
1. `[String]` from `Field::string()->list()->nullable()` according to this PR
2. `[String!]` from `Field::string()->list()->nullable()` according to original implementation
3. `[String]!` not implemented right now.
4. `[String!]!` from `Field::string()->list()`

What is the _actual_ desired outcome? Option `1` would allow the items of the list to be nullable, and the list itself being null too. Option `2` seems somehow more logical to me, however as https://github.com/facebook/graphql/issues/171#issuecomment-228236892 points out, all 4 are important to have. However, as GraphQL is by default *nullable*, "it makes sense that by default the list itself and the items inside are could be null."

Reply:
@erikgaal added support for `1` & `2`. The default used to be `3` but that's `4` now.